plugins/flytekit-pydantic-v2/flytekitplugins/pydantic/transformer.py

```diff
@@ -0,0 +1,74 @@
+import json
+from typing import Type
+
+import msgpack
+from google.protobuf import json_format as _json_format
+
+from flytekit import FlyteContext
+from flytekit.core.constants import MESSAGEPACK
+from flytekit.core.type_engine import TypeEngine, TypeTransformer, TypeTransformerFailedError
+from flytekit.loggers import logger
+from flytekit.models import types
+from flytekit.models.literals import Binary, Literal, Scalar
+from flytekit.models.types import LiteralType, TypeStructure
+from pydantic import BaseModel
+
+
+class PydanticTransformer(TypeTransformer[BaseModel]):
+    def __init__(self):
+        super().__init__("Pydantic Transformer", BaseModel, enable_type_assertions=False)
+
+    def get_literal_type(self, t: Type[BaseModel]) -> LiteralType:
+        schema = t.model_json_schema()
+        literal_type = {}
+        fields = t.__annotations__.items()
+
+        for name, python_type in fields:
+            try:
+                literal_type[name] = TypeEngine.to_literal_type(python_type)
+            except Exception as e:
+                logger.warning(
+                    "Field {} of type {} cannot be converted to a literal type. Error: {}".format(name, python_type, e)
+                )
+
+        ts = TypeStructure(tag="", dataclass_type=literal_type)
+
+        return types.LiteralType(simple=types.SimpleType.STRUCT, metadata=schema, structure=ts)
+
+    def to_literal(
+        self,
+        ctx: FlyteContext,
+        python_val: BaseModel,
+        python_type: Type[BaseModel],
+        expected: types.LiteralType,
+    ) -> Literal:
+        json_str = python_val.model_dump_json()
```

Comment:
Can we add a comment here saying that for pydantic, we have to go through json first?  Just to make it clear.

Reply:
YES of course